compiler/rustc_trait_selection/src/error_reporting/infer/need_type_info.rs

```diff
@@ -613,6 +666,130 @@ impl<'a, 'tcx> TypeErrCtxt<'a, 'tcx> {
             }),
         }
     }
+
+    fn get_suggestions(
+        &self,
+        self_ty: Ty<'tcx>,
+        def_id: DefId,
+        target_type: bool,
+        param_env: ty::ParamEnv<'tcx>,
+        args: Option<&ty::GenericArgs<'tcx>>,
+        predicate: Option<ty::Predicate<'tcx>>,
+    ) -> Vec<String> {
+        let tcx = self.infcx.tcx;
+        let mut paths = vec![];
+        let name = tcx.item_name(def_id);
+        let empty_args = |def_id| {
+            ty::GenericArgs::for_item(tcx, def_id, |param, _| {
+                // We don't want to name the arguments, we just want to give an
+                // idea of what the syntax is.
+                match param.kind {
+                    ty::GenericParamDefKind::Lifetime => tcx.lifetimes.re_erased.into(),
+                    ty::GenericParamDefKind::Type { .. } => self.next_ty_var(DUMMY_SP).into(),
+                    ty::GenericParamDefKind::Const { .. } => self.next_const_var(DUMMY_SP).into(),
+                }
+            })
+        };
+        let args = args.unwrap_or_else(|| empty_args(def_id));
```

Comment:
What is even the point of passing `args` here? Are they not just always filled with garbage?

Reply:
you're right, vestigial from earlier attempts, removing